cmd/injective-price-oracle/oracle.go

```diff
@@ -228,3 +294,34 @@ func oracleCmd(cmd *cli.Cmd) {
 		closer.Hold()
 	}
 }
+
+func ConnectWebSocket(ctx context.Context, websocketUrl, urlHeader string) (conn *websocket.Conn, err error) {
+	u, err := url.Parse(websocketUrl)
+	if err != nil {
+		log.Fatal("Error parsing URL:", err)
+		return &websocket.Conn{}, err
+	}
+
+	header := http.Header{}
+	header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(urlHeader)))
+
+	dialer := websocket.DefaultDialer
+	dialer.EnableCompression = true
+	retries := 0
+	for {
+		conn, _, err = websocket.DefaultDialer.DialContext(ctx, u.String(), header)
+		if err != nil {
```

Comment:
It should retry only in case of no context error
```suggestion
		if ctx.Err() != nil {
			return ctx.Err()
		} else if err != nil {
```